### ISSUE TITLE
Preview version: the Vite plugin crashes server.

### ISSUE THREAD

**Issue by lolmaus** [nubolab-ffwd/svelte-fluent#1242]
When I add the Vite plugin to my Svelte 5 app, it crashes the server like this:

```
ReferenceError: require is not defined in ES module scope, you can use import instead
This file is being treated as an ES module because it has a '.js' file extension and '/workspaces/4eign-takehome/package.json' contains "type": "module". To treat it as a CommonJS script, rename it to use the '.cjs' file extension.
    at requireXMLHttpRequestImpl (file:///workspaces/4eign-takehome/.svelte-kit/output/server/chunks/utils.js:185165:26)
    at file:///workspaces/4eign-takehome/.svelte-kit/output/server/chunks/utils.js:186582:18
    at requireXMLHttpRequest (file:///workspaces/4eign-takehome/.svelte-kit/output/server/chunks/utils.js:186583:5)
    at requireInterfaces (file:///workspaces/4eign-takehome/.svelte-kit/output/server/chunks/utils.js:194261:21)
    at requireWindow (file:///workspaces/4eign-takehome/.svelte-kit/output/server/chunks/utils.js:194774:33)
    at file:///workspaces/4eign-takehome/.svelte-kit/output/server/chunks/utils.js:195595:1
    at ModuleJob.run (node:internal/modules/esm/module_job:262:25)
    at async onImport.tracePromise.__proto__ (node:internal/modules/esm/loader:485:26)
    at async Module.component (file:///workspaces/4eign-takehome/.svelte-kit/output/server/nodes/2.js:6:59)
    at async Promise.all (index 1)
```

**Comment by sfriedel** [comment]
Hi @lolmaus, thank you for the report.

I  can't reproduce this error with a new project from `npm create svelte` and also not with any of my other test projects..
Do you have a reproduction case that you can share?

**Comment by lolmaus** [comment]
@sfriedel Yes, sir.

You can see it on this repo: https://github.com/lolmaus/lolma.us-sveltekit/

It uses a static adapter, and initially I assumed the issue was specific to the static adapter. But I've had the same issue on a Vercel adapter in another project. The other project is private but I can add you if you want.

**Comment by sfriedel** [comment]
@lolmaus I managed to reproduce the error with your repository.

The problem is that vite tries to bundle `jsdom` into the server bundle which does not work.

To fix the error you need to add `jsdom` to your `dependencies` in `package.json` (e.g. with `pnpm add jsdom`).

**Comment by lolmaus** [comment]
Uhm, why do you need `jsdom` in the bundle? Does it go into the frontend? And what does the plugin do in the first place? Your Svelte addon seems to work without it...

**Comment by sfriedel** [comment]
> Uhm, why do you need jsdom in the bundle?

`jsdom` is used for server-side rendering of the [Overlay component](https://svelte-fluent-svelte-5-preview.netlify.app/docs/reference/Overlay/). Overlays allow fluent messages to contain HTML elements and to generate the SSR output, a DOM implementation is needed.

> Does it go into the frontend?

No, only the server bundle imports `jsdom`.

> And what does the plugin do in the first place?

Currently it serves 2 functions:

* It provides SSR support for the `Overlay` component
* It allows you to import `.ftl` files directly using `import resources from 'path/to/messages.ftl'` directly providing you a `FluentResource` instance instead of having to constructing your own from a string imported via vite's `?raw` syntax.

> Your Svelte addon seems to work without it

If you don't plan to use the `Overlay` component and the `.ftl` imports without `?raw` you can safely skip the vite plugin.

**Comment by lolmaus** [comment]
Thank you for kind explanations! All of this should go into docs!

**Comment by sfriedel** [comment]
> Also, can the error be less confusing? I would expect `jsdom: missing package` or something...

I thought about that but there doesn't seem to be a perfect solution. Let me try to explain why:

svelte-fluent declares compatible versions of `jsdom` as a `peerDependency`. Previously, `pnpm` (not sure about the current yarn and npm behavior)  would not automatically install peer dependencies but instead show a (easily missed) warning when the user installs a package with unsatisfied peer dependencies. If users actually saw the missing peer dependency messsage and then added it to their dependencies, everything would be working.

Nowadays `pnpm` automatically installs peer dependencies. This is why you don't see `jsdom: missing package` because it actually gets installed. The problem is that when `jsdom` gets installed this way the build will fail because `vite` will try to bundle `jsdom` which doesn't appear to be possible and leads to the error you reported.

To avoid that bundling problem `jsdom` needs to be added to the `ssr.externals` [vite config](https://vitejs.dev/config/ssr-options#ssr-external). Since it doesn't get bundled into the server bundle then any more, it needs to be installed as a normal (non-dev) dependency of the package. Installing it as a non-dev dependency also causes `svelte-kit` to add it to `ssr.externals`.

Some options I was thinking of for addressing this are:

1. remove `jsdom` from `peerDependencies`. This would avoid the automatic installation and should result with the build failing due to the missing package.
   - the downside here is that the package manager would not warn about incompatible `jsdom` versions any more
2. change the vite plugin to detect if `jsdom` is not installed as dependency and abort the build with a more descriptive error message
3. change the vite plugin so that it always adds `jsdom` to `ssr.externals` which would make the build fail with `Error [ERR_MODULE_NOT_FOUND]: Cannot find package 'jsdom' imported from path/to/.svelte-kit/output/server/entries/pages/_lang_/_layout.ts.js` error message that might be easier to understand
   - this may still be confusing because it doesn't make it clear that `svelte-fluent` causes the dependency on `jsdom` but it may still be better than the current error

My preferred approach currently would be approach 2 but this may get a bit more complicated so I'm thinking about going with approach 3 for the moment.

**Comment by sfriedel** [comment]
> My preferred approach currently would be approach 2 but this may get a bit more complicated so I'm thinking about going with approach 3 for the moment.

I ended up going with approach 2 in the end. If you want to check it out it's available in version `1.0.0-next.2`
